### ISSUE TITLE
One argument mod/ doesn't return modular inverse

### ISSUE THREAD

**Issue by dsemi** [racket/math#33]
According to the [docs](https://docs.racket-lang.org/math/number-theory.html#%28def._%28%28lib._math%2Fnumber-theory..rkt%29._mod%2F%29%29), `mod/` should return the modular inverse of its argument when it's only supplied with one. This appears to be the correct assessment when looking at the [implementation](https://github.com/racket/math/blob/a82f83e22fd94e6af69d41d804b6a7b66f894438/math-lib/math/private/number-theory/modular-arithmetic.rkt#L90).  
However, when running some test calls in the REPL, the modular inverse isn't returned correctly.
Example:
```
# This behaves as expected and returns the inverse of 3 mod 7 (5)
(modular-inverse 3 7)
> 5

# This also behaves as expected
(with-modulus 7 (mod/ 1 3))
> 5

# However, the single argument call doesn't return the inverse
(with-modulus 7 (mod/ 3))
> 1
```

It's not exactly clear to me why this happens, but I think it's something to do with how the code expands or is compiled in modular-arithmetic.rkt.  If I create my own implementation of `mod/` that copies the definition in modular-arithmetic.rkt, it behaves as expected.

```
(define mod-div
  (case-lambda
    [(a)  (modular-inverse a (current-modulus))]
    [(a b)  (define n (current-modulus))
            (modulo (* a (modular-inverse b n)) n)]
    [(a b . cs)  (mod-div a (apply mod* b (car cs) (cdr cs)))]))

(with-modulus 7 (mod-div 3))
> 5
```

**Comment by soegaard** [comment]
The implementation of `with-modulus` is a bit tricky. 
Within `with-modulus` most of the operations are inlined (in order to avoid 
evaluating the expression `(current-modulus)` repeatedly.
The inlined versions are in `modular-arithmetic-base.rkt` and I think I spotted the error 
(testing whether this is correct must wait till tomorrow):

https://github.com/racket/math/blob/master/math-lib/math/private/number-theory/modular-arithmetic-base.rkt#L155

I believe the order of `n` and `a` needs to be swapped on line 155.

```
153 (define-syntax (modular/ stx)
154     (syntax-case stx ()
155      [(_ n a)  (syntax/loc stx (modular-inverse n a))]
156      [(_ n a b ...)  (syntax/loc stx
157                        (modular* n a (modular-inverse* n (modular* n b ...))))]))
```
Thanks for reporting this bug.

**Comment by soegaard** [comment]
Changing line 155 to:

    [(_ n a)  (syntax/loc stx (modular-inverse a n))]

did fix the bug.

**Comment by jbclements** [comment]
Closed by #34.
